pkgs/build-support/writers/default.nix

```diff
@@ -77,7 +77,11 @@ rec {
     }) ''
       ${compileScript}
       ${lib.optionalString strip
-         "${pkgs.binutils-unwrapped}/bin/strip --strip-unneeded $out"}
+          "${lib.getBin buildPackages.bintools-unwrapped}/bin/${buildPackages.bintools-unwrapped.targetPrefix}strip -S $out"}
+      # Sometimes binaries produced for darwin (e. g. by GHC) won't be valid
+      # mach-o executables from the get-go, but need to be corrected somehow
+      # which is done by fixupPhase.
+      ${lib.optionalString pkgs.stdenvNoCC.hostPlatform.isDarwin "fixupPhase"}
```

Comment:
Maybe, yes. The problem is that it strips unconditionally, but will always silently fail because `stdenvNoCC` has no `bintools`. The text will still be displayed, so this may be confusing for users setting `strip` to `false` (for whatever reasons).